tfjs-backend-webgpu/src/kernels/im2col_webgpu.ts

```diff
@@ -44,11 +45,11 @@ export class Im2ColProgram implements WebGPUProgram {
         [this.workPerThread, 1, 1]);
     this.inputShape = inputShape;
     this.convInfo = convInfo;
-    this.shaderKey = `im2col_${convInfo}`;
+    this.shaderKey = `im2col${convInfo.inShape}`;
```

Comment:
There are too many constant values in this shader. I don't think your change is correct.

Reply:
I think the goal  for this PR is to use shape/size as uniform. There are some left const in program, suggest we do this in another pr

Reply:
I mean `convInfo` is not  saved as uniforms in this program. In your change, it will result incorrect result.

Reply:
${convInfo} is an object. I will put all members used as const in shader key.